### PR TITLE
Remove old parfit password login

### PR DESCRIPTION
**Pull request by mordaroso**
It has been more than 1.5 years that users were able to login with their old email/password that was stored on the parfit database.

To clean up the parfit code base (https://github.com/centre-for-effective-altruism/parfit/pull/563) we got rid of the legacy password logic. This PR also removes this legacy login functionality from Auth0.

Users who haven't logged in since we moved to Auth0 can still use the password reset functionality to recreate a new login.

### CODE REVIEW

File: scripts/db/src/login.ts [hunk 1]
```diff
@@ -120,75 +101,11 @@ async function login(
       }
     }
 
-    async function loginParfitUser(): Promise<CallbackUser | null> {
```
Comment (oetherington) [inline scripts/db/src/login.ts:123]: A lot of the code being removed here is also duplicated in `scripts/db/src/get-user.ts` so it probably also needs to be removed there as well. I guess we can remove `pg` as a dependancy in `package.json` too.

**Review by oetherington — APPROVED** [review]
Looks good to me subject to the comment I left.

Also, FYI @jpaddison3, who should also be aware of this PR if he hasn't seen it already.

File: scripts/db/src/login.ts (hunk 1 above)
Comment (jpaddison3) [reply scripts/db/src/login.ts:123]: It actually seems rather important that we *don't* remove it from get-user. IDK, it depends on your point of view. But its presence in get-user prevents users from signing up and seeing someones donation details from a long time ago.

**Review by sarahbaka — APPROVED** [review]

File: scripts/db/src/login.ts (hunk 1 above)
Comment (mordaroso) [reply scripts/db/src/login.ts:123]: Yes, I agree. 
It's important that we keep it for get-user for exactly that reason. Thanks for pointing it out JP

**Pull request merged**